### ISSUE TITLE
Mutex and semaphore-blocked workflows are hidden in the UI

### ISSUE THREAD

**Issue by tico24** [argoproj/argo-workflows#9936]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

Pre 3.4.2 (I think...but definitely pre 3.4.0), when one had workflow(s) blocked by either mutexes or semaphore, they would show on the UI with an appropriate message to say that they are waiting to run.

Now in 3.4.2+, these blocked workflows are hidden until they start running. This is confusing to the casual observer because there's no proof that the workflow is waiting to run.

The new pending counter that appeared in 3.4.2 also doesn't show these workflows as pending. The counter remains at 0.

There seem to be some situations where I can refresh the UI in such a way that some, but not all mutexes will show on the screen. Refreshing the page makes them go away again. 

### Version

3.4.3

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
metadata:
  generateName: mutex-test-
  namespace: argo
  labels:
    example: 'true'
spec:
  arguments:
    parameters:
      - name: message
        value: hello argo
  entrypoint: argosay
  synchronization:
    mutex:
      name: test
  templates:
    - name: argosay
      inputs:
        parameters:
          - name: message
            value: '{{workflow.parameters.message}}'
      container:
        name: main
        image: argoproj/argosay:v2
        command:
          - /argosay
        args:
          - echo
          - '{{inputs.parameters.message}}'
  ttlStrategy:
    secondsAfterCompletion: 300
  podGC:
    strategy: OnPodCompletion
```

### Logs from the workflow controller

n/a

nothing related to mutex/semaphore shown in the controller logs

### Logs from in your workflow's wait container

n/a - pending workflows don't have a wait container

**Comment by sarabala1979** [comment]
@tico24 Can you verify Argo UI is not showing any pending workflow? Please make sure you clear all your filters.

**Comment by tico24** [comment]
That's literally the bug I'm raising. They are not showing (most of the time). There are no filters enabled.

**Comment by isubasinghe** [comment]
Hey @tico24 does this issue look like a replicate of https://github.com/argoproj/argo-workflows/issues/9812 by any chance?

**Comment by tico24** [comment]
@isubasinghe Yes, it appears so. 

I haven't properly tested, but it does look like it might be fixed in v3.4.4

**Comment by tico24** [comment]
@isubasinghe Will do. It'll probably be nest week realistically.

**Comment by tico24** [comment]
This does indeed seem to be fixed in v 3.4.4. Thanks!
